database/dtb.ksy

```diff
@@ -0,0 +1,129 @@
+meta:
+  id: dtb
+  title: Flattened Devicetree Format
+  file-extension: dtb
+  application:
+    - Linux
+    - Das U-Boot
+  xref:
+    wikidata: Q16960371
+  license: CC0-1.0
+  encoding: ASCII
+  endian: be
+doc: |
+  Also referred to as Devicetree Blob (DTB). It is a flat
+  binary encoding of data (primarily devicetree data, although
+  other data is possible as well).
+
+  On Linux systems that support this the blobs can be accessed in
+  /sys/firmware/fdt :
+
+  - https://www.kernel.org/doc/Documentation/ABI/testing/sysfs-firmware-ofw
+
+  The encoding of strings used in the strings block and struct block is
+  actually a subset of ASCII:
+
+  https://github.com/devicetree-org/devicetree-specification/blob/master/source/chapter2-devicetree-basics.rst
+
+  Example files:
+
+  - https://github.com/qemu/qemu/tree/master/pc-bios
+doc-ref:
+  - https://github.com/devicetree-org/devicetree-specification/releases/tag/v0.3
+  - https://github.com/devicetree-org/devicetree-specification/blob/ba2aa67/source/flattened-format.rst
+  - https://elinux.org/images/f/f4/Elc2013_Fernandes.pdf
+seq:
+  - id: magic
+    -orig-id: magic
+    contents: [0xd0, 0x0d, 0xfe, 0xed]
+  - id: total_size
+    -orig-id: totalsize
+    type: u4
+  - id: structure_block_offset
+    -orig-id: off_dt_struct
+    type: u4
+  - id: strings_block_offset
+    -orig-id: off_dt_strings
+    type: u4
+  - id: memory_reservation_block_offset
+    -orig-id: off_mem_rsvmap
+    type: u4
+  - id: version
+    -orig-id: version
+    type: u4
+  - id: last_compatible_version
+    -orig-id: last_comp_version
+    type: u4
+    valid:
+      max: version
+  - id: boot_cpuid_phys
+    -orig-id: boot_cpuid_phys
+    type: u4
+  - id: strings_block_size
+    -orig-id: size_dt_strings
+    type: u4
+  - id: structure_block_size
+    -orig-id: size_dt_struct
+    type: u4
+instances:
+  memory_reservation_block:
+    pos: memory_reservation_block_offset
+    size: structure_block_offset - memory_reservation_block_offset
+  structure_block:
+    pos: structure_block_offset
+    size: strings_block_offset - structure_block_offset
+    type: fdt_block
+  strings_block:
+    pos: strings_block_offset
+    size: strings_block_size
+    type: strings
+types:
+  strings:
+    seq:
+      - id: strings
+        type: strz
+        encoding: ASCII
+        repeat: eos
+  fdt_node:
+    seq:
+      - id: token_type
+        type: u4
+        enum: fdt
+      - id: fdt_node_body
+        type:
+          switch-on: token_type
+          cases:
+            fdt::begin_node: fdt_begin_node
+            fdt::prop: fdt_prop
+  fdt_block:
+    seq:
+      - id: fdt_nodes
+        type: fdt_node
+        repeat: until
+        repeat-until: _.token_type == fdt::end
+  fdt_begin_node:
+    seq:
+      - id: name
+        type: strz
+        encoding: ASCII
```

Comment:
You have put the`encoding: ASCII` into top-level `/meta`, so you can remove it everywhere in the file:

```suggestion
```

There is also another occurrence above this one:

https://github.com/kaitai-io/kaitai_struct_formats/blob/0ea5632153a181f19769958407ff89b9e2cc3789/database/dtb.ksy#L85

Reply:
Yes, sorry, I forgot, currently juggling too many things at the same time.